tests/io/test_curate_validate_records.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+from augur.curate import validate_records
+from augur.errors import AugurError
+
+
+@pytest.fixture
+def good_records():
+    return [
+        {"geo_loc_name": "Canada/Vancouver"},
+        {"geo_loc_name": "Canada/Vancouver"},
+    ]
+
+
+@pytest.fixture
+def bad_records():
+    return [
+        {"geo_loc_name": "Canada/Vancouver"},
+        {"geo_loc_name2": "Canada/Vancouver"},
+    ]
+
+
+class TestCurateValidateRecords:
+    def test_validate_input(self, good_records):
+        validated_records = validate_records(good_records, "test_subcmd", True)
+        assert list(validated_records) == good_records, "good input records validate"
+
+    def test_validate_output(self, good_records):
+        validated_records = validate_records(good_records, "test_subcmd", False)
+        assert list(validated_records) == good_records, "good output records validate"
+
+    def test_validate_bad_records(self, bad_records):
+        try:
+            validate_records(bad_records, "test_subcmd", True)
+        except AugurError as e:
+            assert e.startswith(
+                "Records do not have the same fields!"
+            ), "bad input records throw exception with expected message"
```

Comment:
Pytest has a built in way to of [asserting exceptions](https://docs.pytest.org/en/8.2.x/getting-started.html#assert-that-a-certain-exception-is-raised). 

You can see an example in https://github.com/nextstrain/augur/blob/f9671ac381a6cd64065841776fb874c2a0b88909/tests/io/test_metadata.py#L83-L86

Reply:
ah, right, cool, i always forget about this, will convert over.